kasa/cli.py

```diff
@@ -771,27 +853,30 @@ async def cmd_command(dev: Device, module, child, command, parameters):
 
 
 @cli.command()
-@pass_dev
 @click.option("--index", type=int, required=False)
 @click.option("--name", type=str, required=False)
 @click.option("--year", type=click.DateTime(["%Y"]), default=None, required=False)
 @click.option("--month", type=click.DateTime(["%Y-%m"]), default=None, required=False)
 @click.option("--erase", is_flag=True)
-async def emeter(dev: Device, index: int, name: str, year, month, erase):
-    """Query emeter for historical consumption.
+@click.pass_context
+async def emeter(ctx: click.Context, index, name, year, month, erase):
+    """Query emeter for historical consumption."""
+    logging.warning("Deprecated, use 'kasa energy'")
+    return await ctx.invoke(
+        energy, child_index=index, child=name, year=year, month=month, erase=erase
+    )
 
-    Daily and monthly data provided in CSV format.
-    """
-    if index is not None or name is not None:
-        if not dev.is_strip:
-            error("Index and name are only for power strips!")
-            return
 
-        if index is not None:
-            dev = dev.get_plug_by_index(index)
-        elif name:
-            dev = dev.get_plug_by_name(name)
+@cli.command()
+@click.option("--year", type=click.DateTime(["%Y"]), default=None, required=False)
+@click.option("--month", type=click.DateTime(["%Y-%m"]), default=None, required=False)
+@click.option("--erase", is_flag=True)
+@pass_dev_or_child
+async def energy(dev: Device, year, month, erase):
```

Comment:
This whole function should be modified to work with the module and not be `iot` specific in another PR.